lib/Wrapper/AMDGPU/GCNOptSched.cpp

```diff
@@ -20,18 +21,10 @@ static cl::opt<bool>
                          cl::desc("Limit occpancy target using perf hints."),
                          cl::init(false), cl::Hidden);
 
-static ScheduleDAGInstrs *createOptSchedGCN(MachineSchedContext *C) {
-  ScheduleDAGMILive *DAG = new ScheduleDAGOptSchedGCN(
-      C, llvm::make_unique<GCNMaxOccupancySchedStrategy>(C));
-  DAG->addMutation(createLoadClusterDAGMutation(DAG->TII, DAG->TRI));
-  DAG->addMutation(createStoreClusterDAGMutation(DAG->TII, DAG->TRI));
-  return DAG;
-}
-
 // Register the machine scheduler.
-static MachineSchedRegistry
-    OptSchedMIRegistry("gcn-optsched", "Use the GCN OptSched scheduler.",
-                       createOptSchedGCN);
+// static MachineSchedRegistry
+//    OptSchedGCNMIRegistry("gcn-optsched", "Use the GCN OptSched scheduler.",
+//                       createOptSchedGCN);
```

Comment:
Why do we no longer need this?

Reply:
The registration is done in GCNOptSchedReg.h, I was not able to get LLVM to pick up on OptSched unless it was explicitly included from an LLVM source file. Due to this, I pulled the registration code into a header file which I included in AMDGPUTargetMachine.cpp. Do you know a way around this for static builds?

Reply:
Ah, this is probably the thing where the linker sees that it's unused so it skips it. I found this article: https://www.cppstories.com/2018/02/static-vars-static-lib/. Sounds like you can work around it by doing one of these things (or other options that I didn't think of):

 - In LLVM, either find some way to refer to these static variables, or call a registration function.
 - Use some sort of `-Wl,--whole-archive` flag, which tells the linker not to cut out unused parts of the static library. I have no experience with this.
 - Use an `OBJECT` library rather than a `STATIC` library. This is a CMake-only concept. An OBJECT library is just telling CMake about the bundle of files, so there's no static library generated; linking with it will just link the `.o` files together without archiving (IIUC).

Reply:
Thanks for looking into this! This seems like a cool / fun exercise -- and has the benefit of making our project more extensible. At the present, I think this qualifies as a separate issue rather than a required piece of this PR -- I'll open an issue with this. Thanks.